examples/langDefs-resources/template.lang

```diff
@@ -0,0 +1,365 @@
+                                 ESCAPE SEQUENCES                               
+  ------------------------------------------------------------------------------
+  If the language at hand supports escape sequences, define a RegEx pattern to
+  capture them.
+
+    -- https://en.wikipedia.org/wiki/Escape_sequences_in_C
+
+  NOTE: Escape sequences are not restricted to occur inside strings only, they
+        will be matched anywhere in the source code (some languages, like Perl
+        and Bash, allow their use anywhere). Usually this doesn't constitute a
+        problem, but in some languages this uncostrained behaviour might cause
+        false positives matches; in such cases you'll need to restrict escape
+        sequences occurence to inside-strings context only by implementing a
+        custom hook via the OnStateChange() function --- see "Hook Preset #01"
+        further down.
+--]]
+  -- Highlight's default built-in Escape definition:
+  Escape=[=[ \\u[[:xdigit:]]{4}|\\\d{3}|\\x[[:xdigit:]]{2}|\\[ntvbrfa\\\?'"] ]=],
+--[[------------------------------------------------------------------------------
+                                  INTERPOLATION                                 
+    ------------------------------------------------------------------------------
+    String, regex of interpolation sequences (optional)
+
+    To understand interpolation, here is an example from Javascript:
+
+        var apples = 6;
+        console.log(`There are ${apples} apples in the basket!`);
+        
+    which will otuput:
+
+        There are 6 apples in the basket!
+
+    References:
+    -- https://en.wikipedia.org/wiki/String_interpolation
+--]]
+  Interpolation=[=[ \$\{.+?\} ]=], -- Javascript Interpolation: ${ ... }
+}
+
+
+--[[==============================================================================
+                                      OPERATORS                                   
+    ==============================================================================
+--]]
+Operators=[=[ \&|<|>|\!|\||\=|\/|\*|\%|\+|\-|~ ]=] -- Match: &<>!|=/*%+-~ 
+
+--[[==============================================================================
+                                        DIGITS                                    
+    ==============================================================================
+    String, Regular expression which defines digits (optional).
+--]]
+-- Highlight's default built-in Digits definition:
+Digits=[=[ (?:0x|0X)[0-9a-fA-F]+|\d*[\.]?\d+(?:[eE][\-\+]\d+)?[lLuU]* ]=]
+
+--[[==============================================================================
+                                       KEYWORDS                                   
+    ==============================================================================
+    Keywords = { Id, List|Regex, Group? }
+
+      Id:    Integer, keyword group id (values 1-4, can be reused for several keyword
```

Comment:
Actually the number is not limited to 4, but the packaged themes support 4 groups. So if a keyword group 5 is defined, the theme also needs to be enhanced.

Reply:
True, I forgot to add a note about that!

Added with commit 0a1595f